src/kudu/consensus/consensus_peers.cc

```diff
@@ -243,6 +243,7 @@ void Peer::SendNextRequest(bool even_if_queue_empty) {
       request_.committed_index() : kMinimumOpIdIndex;
 
   if (PREDICT_FALSE(!s.ok())) {
+    failed_attempts_++;
```

Comment:
Leave a small comment here that setting "failed_attempts_" here ensures that this peer degenerates into "send-only-heartbeat" and skip sending during regular writes.